sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/linearRegression.scala

```diff
@@ -105,3 +106,100 @@ case class RegrAvgY(left: Expression, right: Expression)
       newLeft: Expression, newRight: Expression): RegrAvgY =
     this.copy(left = newLeft, right = newRight)
 }
+
+abstract class Regression
+  extends DeclarativeAggregate with ImplicitCastInputTypes with BinaryLike[Expression] {
+
+  override def nullable: Boolean = true
+  override def dataType: DataType = DoubleType
+  override def inputTypes: Seq[AbstractDataType] = Seq(DoubleType, DoubleType)
+
+  protected val count = AttributeReference("count", DoubleType)()
+  protected val meanX = AttributeReference("meanX", DoubleType)()
+  protected val meanY = AttributeReference("meanY", DoubleType)()
+  protected val c2 = AttributeReference("c2", DoubleType)()
+  protected val m2X = AttributeReference("m2X", DoubleType)()
+
+  override val aggBufferAttributes: Seq[AttributeReference] = Seq(count, meanX, meanY, c2, m2X)
+
+  override val initialValues: Seq[Expression] = Array.fill(5)(Literal(0.0))
+
+  override lazy val updateExpressions: Seq[Expression] = {
```

Comment:
do you follow any reference implementation? This doesn't seem like following https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance

Reply:
REGR_SLOPE(y,x) = COVAR_POP(y,x) / VAR_POP(x)
COVAR_POP(y,x) follows 
![image](https://user-images.githubusercontent.com/8486025/157395844-e6080828-bfc9-48a6-afe7-f58d7732f450.png)
and we can reference https://github.com/apache/spark/blob/52e7602344036a649bb19ed642dbd74bc7ee9cb1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Covariance.scala#L72